platforms/software/maven/src/main/java/org/gradle/api/publish/maven/tasks/GenerateMavenPom.java

```diff
@@ -45,112 +45,57 @@
 @UntrackedTask(because = "Gradle doesn't understand the data structures used to configure this task")
 public abstract class GenerateMavenPom extends DefaultTask {
 
-    private final Transient.Var<MavenPom> pom = varOf();
-    private Object destination;
-    private final Cached<MavenPomFileGenerator.MavenPomSpec> mavenPomSpec = Cached.of(() ->
-        MavenPomFileGenerator.generateSpec((MavenPomInternal) getPom())
-    );
-
-    @Inject
-    protected FileResolver getFileResolver() {
-        throw new UnsupportedOperationException();
-    }
-
-    /**
-     * Get the version range mapper.
-     *
-     * @deprecated This method will be removed in Gradle 9.0
-     */
-    @Inject
-    @Deprecated
-    protected VersionRangeMapper getVersionRangeMapper() {
-        throw new UnsupportedOperationException();
-    }
-
-    /**
-     * The values set by this method are ignored.
-     *
-     * @deprecated This method will be removed in Gradle 9.0.
-     */
-    @Deprecated
-    public GenerateMavenPom withCompileScopeAttributes(ImmutableAttributes compileScopeAttributes) {
-
-        DeprecationLogger.deprecateMethod(GenerateMavenPom.class, "withCompileScopeAttributes(ImmutableAttributes)")
-            .withContext("This method was never intended for public use.")
-            .willBeRemovedInGradle9()
-            .withUpgradeGuideSection(8, "generate_maven_pom_method_deprecations")
-            .nagUser();
-
-        return this;
-    }
-
-    /**
-     * The values set by this method are ignored.
-     *
-     * @deprecated This method will be removed in Gradle 9.0.
-     */
-    @Deprecated
-    public GenerateMavenPom withRuntimeScopeAttributes(ImmutableAttributes runtimeScopeAttributes) {
-
-        DeprecationLogger.deprecateMethod(GenerateMavenPom.class, "runtimeScopeAttributes(ImmutableAttributes)")
-            .withContext("This method was never intended for public use.")
-            .willBeRemovedInGradle9()
-            .withUpgradeGuideSection(8, "generate_maven_pom_method_deprecations")
-            .nagUser();
+    private final Transient<Property<MavenPom>> pom = Transient.of(getObjectFactory().property(MavenPom.class));
 
-        return this;
-    }
+    private final Provider<MavenPomFileGenerator.MavenPomSpec> mavenPomSpec = getPom().map(pom ->
+        MavenPomFileGenerator.generateSpec((MavenPomInternal) pom)
+    );
 
     /**
      * The Maven POM.
-     *
-     * @return The Maven POM.
      */
     @Internal
-    @ToBeReplacedByLazyProperty
-    public MavenPom getPom() {
+    @ReplacesEagerProperty
+    public Property<MavenPom> getPom() {
```

Comment:
❓ Should this be a `Property`? It looks like a nested type
/cc @jvandort

Reply:
This is similar to the repository thing. We really want a reference to the POM configured on the publication, but we don't have a way to do that. 

Ideally the POM on this task would be immutable, and we could have some way of getting an immutable/read-only reference to the POM on the publication. Something like

```
tasks.register("generatePomFile", GenerateMavenPom) {
    pom = publications.named("foo").map { it.pom.asImmutable }
}
```

Where `asImmutable` either returned a property-less version of POM (with all static data), or finalized all properties of the original type

Without some pattern like this, I think we probably have to leave this as-is. Migrating to nested would mean manually coping all the data from one instance to another